### PR TITLE
NXP backend: Enable `permute_copy` with new Neutron MLIR flow.

### PR DESCRIPTION
**Pull request by MartinPavella**
### Summary
This PR updates the support for the `permute_copy` operator in the NXP backend to reflect the requirements of the new Neutron MLIR flow. In short, Neutron now supports all possible permutations without any restrictions.

### Test plan
Unit tests provided.

cc @robert-kalmar @JakeStevens @digantdesai @rascani

### CODE REVIEW

**Comment by MartinPavella** [comment]
The PR is based on https://github.com/pytorch/executorch/pull/19938. Until that one is merged, this one will stay as draft, and the [pull / unittest-nxp-neutron / linux-job (pull_request)](https://github.com/pytorch/executorch/actions/runs/26887692655/job/79304497096?pr=19974) will not pass (locally, the tests are passing).

@novak-vaclav, @irtrukhina, please feel free to review as well.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py [hunk 1]
```diff
@@ -62,11 +57,15 @@ def get_supported_transpositions(
     output_shape = node.meta["val"].shape
     perm = list(node.args[1])
 
-    to_nchw_perm = translator.create_channels_last_to_channels_first_permutation(
-        len(input_shape), True
+    to_nchw_perm = list(
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py:60]: nit: I am confused why you are setting `return_list=True` arg in the `create_channels_last_to_channels_first_permutation`/`create_channels_first_to_channels_last_permutation` and then still converting the result to `list`. In such case, setting `return_list=True` is unnecessary.

If you keep the `return_list=True`, please name the argument in the method call explicitly to improve readability, ie. `translator.create_channels_last_to_channels_first_permutation(len(input_shape), return_list=True)`

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py [hunk 2]
```diff
@@ -62,11 +57,15 @@ def get_supported_transpositions(
...
+        translator.create_channels_last_to_channels_first_permutation(
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py]: nit: As I am looking at the code, it took me a while to understand the merging logic and what `separate_pre`, `merged_pre` etc mean.
I would add some kind of a sketch illustrating what the `separate_pre`, `merged_pre` etc signify. Something like what we have in `aten` passes docstrings.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py [hunk 3]
```diff
@@ -222,6 +221,11 @@ def _handle_formatless_input_and_channels_first_output(
     def _handle_channels_first_input_and_output(
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py]: nit: the name of this method seems counterintuitive because it handles something, but at the same time returns something else.
I would suggest renaming it like `get_main_perm_with_channels_first_format`.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py [hunk 4]
```diff
@@ -249,7 +253,7 @@ def _handle_channels_first_input_and_output(
                     t_op, 0, perm_dict["separate_pre"]["perm"]
                 )
             )
-            perm = perm_dict["everything_merged"]["supported"]
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py:252]: Was this a discovered bug or is the change necessary to enabled `permute_copy` using MLIR flow?

File: backends/nxp/tests/ir/converter/node_converter/test_permute_copy_converter.py
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_permute_copy_converter.py]: nit: please add 2026

**Review by novak-vaclav — APPROVED** [review]
I suggest only small details to improve readability since I am seeing this part of the code base for the first time.

Otherwise very nice! 😊 LGTM

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py (hunk 1 above)
Comment (MartinPavella) [reply backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py:60]: I added the conversion to list because the type-checker in my IDE was complaining (`create_channels_last_to_channels_first_permutation` returns `np.ndarray | list[int]`, and the type-checker doesn't know it's always `list[int]` when `return_list=True` is used).

Good point though. I will keep the conversion to list and I will remove the `return_list=True` argument.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py (hunk 4 above)
Comment (MartinPavella) [reply backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py:252]: It was a discovered bug.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py (hunk 2 above)
Comment (MartinPavella) [reply backends/nxp/backend/ir/converter/node_converters/ops_converters/permute_copy_converter.py]: Hmm, good idea. That would definitely be useful.
I'm just thinking if there is time to add it right now. This PR (alongside the `cat` and `softmax` PRs) is necessary to enable QuartzNet for Jake, and it should be merged ASAP.

I will leave it as is for now. We can add the illustrations later. Thanks for the suggestion though 👍🏻

File: backends/nxp/backend/edge_helper.py [hunk 5]
```diff
@@ -108,18 +122,14 @@ def try_get_tensor_constant_from_node(
 
 
 def _is_dequantize(node_: Node) -> bool:
-    return node_.op == "call_function" and node_.target in [
-        exir_ops.edge.quantized_decomposed.dequantize_per_tensor.default,
-        exir_ops.edge.quantized_decomposed.dequantize_per_channel.default,
+    return node_.op == "call_function" and node_.target in DEQUANTIZE_OPERATORS + [
         torch.ops.quantized_decomposed.dequantize_per_tensor.default,
         torch.ops.quantized_decomposed.dequantize_per_channel.default,
```
Comment (robert-kalmar) [inline backends/nxp/backend/edge_helper.py]: Why we do not include these ops into DEQUANTIZE_OPERATORS? Same for QUANTIZE_OPERATORS.

File: backends/nxp/tests/ir/converter/node_converter/test_mean_dim_converter.py [hunk 6]
```diff
@@ -201,11 +201,33 @@ def test__no_reduction__keepdim_false__delegated(self, mocker, input_shape, dim)
         [((1, 7, 3, 3), 1)],
         ids=lambda val: f"shape={val}" if isinstance(val, tuple) else f"dim={val}",
     )
-    def test__channels_first(self, mocker, input_shape, dim, keep_dim):
+    def test__channels_first__keep_dim__true(self, mocker, input_shape, dim):
```
Comment (robert-kalmar) [inline backends/nxp/tests/ir/converter/node_converter/test_mean_dim_converter.py]: Minor: I thing, you can fuse those 2 tests right away, and just use the `pytest.param`'s `marks` attribute mark the failing case (`keep_dim = False`).
```suggestion
   @pytest.mark.parametrize(
       "keep_dim", 
       [
          pytest.param(True, id="keep_dim_true." ), 
          pytest.param(False, id="keep_dim_false.", marks=pytest.mark.xfail(strict=True, reason="Known format inference bug (EIEX-937)), 
          ]
    def test__channels_first__keep_dim__true(self, mocker, input_shape, dim, keep_dim):
    ...
```

**Review by robert-kalmar — APPROVED** [review]

File: backends/nxp/backend/edge_helper.py (hunk 5 above)
Comment (MartinPavella) [reply backends/nxp/backend/edge_helper.py]: I'm not sure. I didn't want to change the `QUANTIZE_OPERATORS` in case something relies on the current state. But I guess if adding the `torch.ops.` quantize ops would break something, it would be a bug that we need to fix. 
I will add the 2 ops to the `[DE]QUANTIZE_OPERATORS`.

File: backends/nxp/tests/ir/converter/node_converter/test_mean_dim_converter.py (hunk 6 above)
Comment (MartinPavella) [reply backends/nxp/tests/ir/converter/node_converter/test_mean_dim_converter.py]: I wanted to explicitly separate the failing test.
But I guess doing it your way will remove the need to modify 2 tests once it is fixed. I will use your suggestion.

**Comment by MartinPavella** [comment]
The failing checks seem unrelated and the [internal CI](https://bamboo3.sw.nxp.com/browse/MLTECE-EXIGH179-5) is passing. Merging.
